simtools/layout/layout_array.py

```diff
@@ -297,11 +299,35 @@ def _initialize_coordinate_systems(self, center_dict=None):
 
         """
 
+        center_dict = {} if center_dict is None else center_dict
+
         self._array_center = TelescopePosition()
         self._array_center.name = "array_center"
         self._array_center.set_coordinates("corsika", 0.0 * u.m, 0.0 * u.m, 0.0 * u.m)
+        self._set_array_center_mercator(center_dict)
+        self._set_array_center_utm(center_dict)
+        try:
+            self._array_center.set_altitude(u.Quantity(center_dict.get("center_alt", np.nan * u.m)))
+        except TypeError:
+            pass
+        try:
+            _name = center_dict.get("array_name")
+            self.name = _name if _name is not None else self.name
+        except KeyError:
```

Comment:
Don't think you can get a KeyError when using `get`. It will just return `None` without raising an exception so the `try` is not necessary (no test for it either).

Reply:
Yes, correct!